File: app/src/main/java/com/akari/ppx/xp/hook/code/auto/BrowseHook.java
package com.akari.ppx.xp.hook.code.auto;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.akari.ppx.xp.hook.code.SuperbHook;

import de.robv.android.xposed.XC_MethodHook;

import static com.akari.ppx.common.constant.Prefs.AUTO_BROWSE_ENABLE;
import static com.akari.ppx.common.constant.Prefs.AUTO_BROWSE_VIDEO_MODE;
import static de.robv.android.xposed.XposedHelpers.callMethod;
import static de.robv.android.xposed.XposedHelpers.getObjectField;

public class BrowseHook extends SuperbHook {
	private static boolean autoBrowse = true;
	private static Object viewPager;

	public static boolean getAutoBrowse() {
		return autoBrowse;
	}

	public static void setAutoBrowse(boolean autoBrowse) {
		BrowseHook.autoBrowse = autoBrowse;
	}

	public static Object getViewPager() {
		return viewPager;
	}

	@Override
	protected void onHook(ClassLoader cl) {
		//ui.j
		//ui.DetailPagerFragment
		hookMethod(AUTO_BROWSE_ENABLE, "com.sup.android.detail.ui.DetailPagerFragment", "onCreateView", LayoutInflater.class, ViewGroup.class, Bundle.class, new XC_MethodHook() {
			@Override
			protected void afterHookedMethod(MethodHookParam param) {
				viewPager = getObjectField(param.thisObject, "A");
			}

		});
		hookMethod(AUTO_BROWSE_VIDEO_MODE, "com.sup.superb.video.viewholder.a", "onPlayerStateChanged", int.class, new XC_MethodHook() {
			@Override
			protected void afterHookedMethod(MethodHookParam param) {
				//3.5.0: ab
				//3.5.1: P
				Object cell = callMethod(param.thisObject, "P");
				if (cell != null && (int) callMethod(cell, "getCellType") == 1 && (int) param.args[0] == 5) {
					Object viewPager = BrowseHook.viewPager;
					if (viewPager != null) {
						callMethod(viewPager, "postDelayed", new ItemThread(viewPager), 0);
					}
				}
			}
		});
	}
}

File: app/src/main/java/com/akari/ppx/xp/hook/auto/BrowseHook.kt
@file:Suppress("unused")

package com.akari.ppx.xp.hook.auto

import android.os.Bundle
import android.view.LayoutInflater
import android.view.ViewGroup
import com.akari.ppx.data.XPrefs
import com.akari.ppx.utils.*
import com.akari.ppx.xp.Init.cl
import com.akari.ppx.xp.Init.detailPagerFragClass
import com.akari.ppx.xp.hook.BaseHook

class BrowseHook : BaseHook {
    override fun onHook() {
        val (isAutoBrowse, isDelayHandoff) = listOf<Boolean>(
            XPrefs("auto_browse"),
            XPrefs("video_delay_handoff")
        )
        isAutoBrowse.check(true) {
            detailPagerFragClass?.hookAfterMethod(
                "onCreateView",
                LayoutInflater::class.java,
                ViewGroup::class.java,
                Bundle::class.java
            ) { param ->
                detailPagerFragClass?.declaredFields?.find { f ->
                    f.type.name == "com.bytedance.ies.uikit.viewpager.SwipeControlledViewPager"
                }?.name.let { f ->
                    viewPager = param.thisObject.getObjectField(f)
                }
            }
        }
        isDelayHandoff.check(true) {
            val videoHolderClass = "com.sup.superb.video.viewholder.a".findClass(cl)
            videoHolderClass.hookAfterMethod("a_", Int::class.java) { param ->
                val feedCell = param.thisObject.callMethod("P") ?: return@hookAfterMethod
                if (feedCell.callMethodAs<Int>("getCellType") == 1 && 5 == param.args[0] as Int)
                    viewPager?.callMethod("postDelayed", object : Runnable {
                        override fun run() {
                            viewPager?.apply {
                                callMethod(
                                    "setCurrentItem",
                                    callMethodAs<Int>("getCurrentItem") + 1
                                )
                            }
                        }
                    }, 0)
            }
        }
    }

    companion object {
        var viewPager: Any? = null
        var isAuto = true
    }
}